src/System.Runtime.Extensions/tests/System/Environment.ProcessorCount.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Runtime.Extensions.Tests
+{
+    public class EnvironmentProcessorCount
+    {
+        [Fact]
+        public void ProcessorCountTest()
+        {
+            //arrange
+            int expected;
+
+            if(Interop.IsWindows)
+            {
+                SYSTEM_INFO sysInfo = new SYSTEM_INFO();
+                SYSTEM_INFO.GetSystemInfo(ref sysInfo);
+                expected = sysInfo.dwNumberOfProcessors;
+            }
+            else
+            {
+                int _SC_NPROCESSORS_ONLN = Interop.IsOSX ? 58 : 84;
```

Comment:
Got the right value for _SC_NPROCESSORS_ONLN from - https://github.com/freebsd/freebsd/blob/master/include/unistd.h